engines/director/lingo/xlibs/popupmenuxobj.cpp

```diff
@@ -148,26 +148,89 @@ void PopUpMenuXObj::close(int type) {
 	}
 }
 
-
 PopUpMenuXObject::PopUpMenuXObject(ObjectType ObjectType) :Object<PopUpMenuXObject>("PopMenu") {
 	_objType = ObjectType;
 }
 
 void PopUpMenuXObj::m_new(int nargs) {
-	g_lingo->printSTUBWithArglist("PopUpMenuXObj::m_new", nargs);
-	g_lingo->dropStack(nargs);
+	PopUpMenuXObject *me = static_cast<PopUpMenuXObject *>(g_lingo->_state->me.u.obj);
+	Datum menuId = g_lingo->pop();
+	Datum menuList = g_lingo->pop();
+
+	me->menu = new Graphics::MacPopUp(menuId.u.i, g_director->_wm->getScreenBounds(), g_director->_wm, menuList.u.s->c_str());
```

Comment:
I would advise against keeping this pointer and rather have `menuId` persisted. Then query WM on the menu.